client/state/site-settings/exporter/reducers.js

```diff
@@ -34,6 +35,40 @@ export function selectedPostType( state = null, action ) {
 	return state;
 }
 
+const postTypeField = ( state = '', action ) => {
+	switch ( action.type ) {
+		case EXPORT_POST_TYPE_FILTER_SET:
+			return Object.assign( {}, state, {
+				[ action.fieldName ]: action.value
+			} );
+	}
+	return state;
+};
+
+const postTypes = ( state = { post: {}, page: {} }, action ) => {
+	switch ( action.type ) {
+		case EXPORT_POST_TYPE_FILTER_SET:
+			return Object.assign( {}, state, {
+				[ action.postType ]: postTypeField( state[ action.postType ], action )
+			} );
+	}
+	return state;
+};
+
+export function selectedAdvancedSettings( state = {}, action ) {
+	switch ( action.type ) {
+		case EXPORT_POST_TYPE_FILTER_SET:
+			return Object.assign( {}, state, {
+				[ action.siteId ]: postTypes( state[ action.siteId ], action )
+			} );
+		case SERIALIZE:
+			return {};
+		case DESERIALIZE:
+			return {};
```

Comment:
Might want to check with @gwwar here - I think if there's nothing to serialize, we just leave these out, right?